# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/WindowGlobalActors.webidl
# Commit: 1cc86398ced0
# Full Hash: 1cc86398ced034fafc66d94799f91221420e1229
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-02-18 04:16:06
# Regressor Bug: 1692350
# File Overlap Count: 1
# Description:
#   Bug 1692350 - Implement a telemetry probe to collect page load data for documents that use lazyload r=emilio,chutten
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104912
# ==============================================================================

diff -r 1ca4c3365fae -r 1cc86398ced0 dom/chrome-webidl/WindowGlobalActors.webidl
--- a/dom/chrome-webidl/WindowGlobalActors.webidl	Wed Feb 17 22:02:17 2021 +0000
+++ b/dom/chrome-webidl/WindowGlobalActors.webidl	Wed Feb 17 22:25:02 2021 +0000
@@ -25,6 +25,10 @@
 
   // True if the principal of this window is for a local ip address.
   readonly attribute boolean isLocalIP;
+
+  // True if the corresponding document has `loading='lazy'` images;
+  // It won't become false if the image becomes non-lazy.
+  readonly attribute boolean hadLazyLoadImage;
 };
 
 // Keep this in sync with nsIContentViewer::PermitUnloadAction.